### PR TITLE
Python C API for extend method

### PR DESCRIPTION
**Pull request by benjeffery**
Python C API work for #1221

### CODE REVIEW

**Comment by benjeffery** [comment]
@jeromekelleher I think this is ready now, note that #1344 should be merged first.

File: python/_tskitmodule.c [hunk 1]
```diff
@@ -1286,6 +1335,10 @@ static PyMethodDef IndividualTable_methods[] = {
         .ml_meth = (PyCFunction) IndividualTable_truncate,
         .ml_flags = METH_VARARGS,
         .ml_doc = "Truncates this table to the specified number of rows." },
+    { .ml_name = "extend",
+        .ml_meth = (PyCFunction) IndividualTable_extend,
+        .ml_flags = METH_VARARGS | METH_KEYWORDS,
+        .ml_doc = "Extend this table from another using optional specifed row_indexes" },
```
Comment (jeromekelleher) [inline python/_tskitmodule.c:1341]: row indexes aren't optional now, are they?

File: python/tests/test_lowlevel.py [hunk 2]
```diff
@@ -417,6 +466,37 @@ def yield_tables(self, ts):
...
+            ll_table.extend(ll_table_copy, row_indexes=[-1])
+        with pytest.raises(_tskit.LibraryError, match="out of bounds"):
+            ll_table.extend(ll_table_copy, row_indexes=[1000])
+        with pytest.raises(_tskit.LibraryError, match="out of bounds"):
+            ll_table.extend(ll_table_copy, row_indexes=range(10000000, 10000001))
+
+        # Uncastable types
+        for dtype in [np.uint32, np.int64, np.uint64, np.float32, np.float64]:
+            with pytest.raises(TypeError, match="Cannot cast"):
+                ll_table.extend(ll_table_copy, row_indexes=np.array([0], dtype=dtype))
```
Comment (jeromekelleher) [inline python/tests/test_lowlevel.py:498]: Good to add a few other types of bad array, like ``[[0, 1], [2, 3]]``, ``[None]``, etc.

**Review by jeromekelleher — APPROVED** [review]
Looks great! Minor comments.

File: python/tests/test_lowlevel.py (hunk 2 above)
Comment (benjeffery) [reply python/tests/test_lowlevel.py:503]: Yeah, good idea!

**Pull request merged**